src/django_components/component.py

```diff
@@ -318,7 +325,6 @@ def __init__(
 
         self.registered_name: Optional[str] = registered_name
         self.outer_context: Context = outer_context or Context()
-        self.component_id = component_id or gen_id()
```

Comment:
Previously, there was a not-yet-documented `Component.component_id`, whereas the ID was unique to every instance of class `Component`.

Now, the ID is:
1. Available under `Component.id`
2. Available only during render
3. It is unique for every time we call render

Thus, when we have a forloop:

```django
{% for ... %}
  {% component "table" / %}
{% endfor %}
```

Then each time we iterate over the loop, the component being rendered will have a different ID.

This is important, because each iteration could yield different JS or CSS variables. And so we may need to distinguish between the different renders, and we use the ID for that.

E.g. the resulting HTML may look like this:
```html
<table data-djc-id-1b3caf>
  ...
</table>

<table data-djc-id-7dacf2>
  ...
</table>

<table data-djc-id-2bc3d4>
  ...
</table>
```

Reply:
Yeah, for CSS I do it that way, that if the component doesn't define any CSS variables, then there's no need for `data-djc-css-...` attribute.

The `data-djc-id-...` attribute is needed though. Because I'm using it also in the CSS scoping feature to avoid applying the scoping attribute to nested components.

E.g. if I have a component like this

```py
class MyTable(Component):
    template = """
        <h1>
          <strong> Hi! </strong>
        </h1>
        <div class="x">
          {% component "button" / %}
        </div>
    """
    css = """
        .x {
          color: red;
        }
    """
    css_scoped = True
```

Then the insides of the template will be transformed to:

```html
<h1 data-djc-scope-1b2c3d>
  <strong data-djc-scope-1b2c3d> Hi! </strong>
</h1>
<div class="x" data-djc-scope-1b2c3d>
 ...
</div>
```

BUT we don't want to apply `data-djc-scope-1b2c3d` to the insides of `{% component "button" / %}`.

Since I know that all components will have the `data-djc-id-xxx`, then for the CSS scoping, I walk down the HTML tree, and keep applying `data-djc-scope-1b2c3d` until I come across `data-djc-id`, in which case I stop walking down that branch. So the final HTML may then look like this:

```html
<h1 data-djc-id-12345 data-djc-scope-1b2c3d>
  <strong data-djc-scope-1b2c3d> Hi! </strong>
</h1>
<div class="x" data-djc-id-12345 data-djc-scope-1b2c3d>
  <!-- START OF {% component "button" / %} -->
  <button data-djc-id-efefef>
    <strong>Click me!</strong>
  </button>
  <!-- END OF {% component "button" / %} -->
</div>